src/precompile.jl

```diff
@@ -0,0 +1,101 @@
+# Precompile statements generated using SnoopCompile, via:
+#   julia> using SnoopCompile, Test, FileIO
+#   julia> tinf = @snoopi_deep begin include("test/rw.jl"); include("test/loadsave.jl"); end
+#   julia> ttot, pcs = ttot, pcs = SnoopCompile.parcel(tinf; tmin=0.05)
+#   julia> SnoopCompile.write("tmp", pcs)
+# With the cutoff at tmin = 0.05, this accounts for about 40% of overall inference time
+# spent in executing /test/rw.jl and /test/loadsave.jl
+# A single CodecZlib argument call was filtered out manually.
+
+function _precompile_()
+    ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
+
+    Base.precompile(Tuple{typeof(write),JLDFile{MmapIO},String,Float64})   # time: 0.8394732
+    Base.precompile(Tuple{typeof(write),JLDFile{MmapIO},String,Base.ReshapedArray{Int64, 2, UnitRange{Int64}, Tuple{}}})   # time: 0.5689673
+    Base.precompile(Tuple{Core.kwftype(typeof(jldopen)),NamedTuple{(:compress,), Tuple{Bool}},typeof(jldopen),String,Bool,Bool,Bool,Type{MmapIO}})   # time: 0.3199544
+    Base.precompile(Tuple{Core.kwftype(typeof(write)),NamedTuple{(:compress,), Tuple{Nothing}},typeof(write),Group{JLDFile{MmapIO}},String,Base.ReshapedArray{Int64, 2, UnitRange{Int64}, Tuple{}},JLDWriteSession{Dict{UInt64, RelOffset}}})   # time: 0.2051841
```

Comment:
Looks like this package tests 32bit as well as 64bit, and sadly SnoopCompile doesn't handle this (sorry). What about just pasting your test script in after the `ccall` in this function? It should be robust, as long as the side-effect of writing a file is not problematic. Alternatively, replace `Int64` with `Int` in various places?

Reply:
Ah, good point! Since the test files I ran to generate these need CodecZlib (which is not a direct dependency of JLD2) I think your second suggestion would be easier - so I just did that (and swapped `UInt64` as well).

PS. This PR is a direct consequence of wanting to try out what I learned from yours and Shuhei's recent workshop - so many thanks from me as well.

Reply:
Ah yes, now that I look more carefully you're not saving a single file, you're running tests, and that takes long enough that it's better to write out the precompile directives.